src/components/helpers/choices.js

```diff
@@ -249,12 +249,18 @@ export default createReactClass({
       const rect = node.getBoundingClientRect();
       const top = rect.top;
       const windowHeight = window.innerHeight;
-      const height = windowHeight - top;
-      if (height !== this.state.maxHeight) {
+      const propsMaxHeight =
+        this.props.field && this.props.field.meta
+          ? this.props.field.meta.choicesMaxHeight
+          : 1;
+      const clampedHeight = Math.min(Math.abs(propsMaxHeight), 1);
+      const actualHeight = windowHeight - top;
+      const desiredMaxHeight = actualHeight * clampedHeight;
+      if (actualHeight !== desiredMaxHeight) {
```

Comment:
I wonder if we'll get weird rounding errors, where `actualHeight !== desiredMaxHeight` when really they are the same, because of JS weirdness with floating points? If so we can round both values before comparing.

Reply:
👍 Great catch!